core/src/test/scala/org/apache/spark/rdd/RDDSuite.scala

```diff
@@ -1298,18 +1298,18 @@ class SizeBasedCoalescer(val maxSize: Int) extends PartitionCoalescer with Seria
       val splitSize = fileSplit.getLength
       if (currentSum + splitSize < maxSize) {
         addPartition(partition, splitSize)
-        index += 1
-        if (index == partitions.size) {
-          updateGroups
-        }
       } else {
-        if (currentGroup.partitions.size == 0) {
+        if (currentGroup.partitions.isEmpty) {
           addPartition(partition, splitSize)
-          index += 1
         } else {
-          updateGroups
+          updateGroups()
+          addPartition(partition, splitSize)
         }
       }
+      index += 1
+      if (index == partitions.length) {
+        updateGroups()
```

Comment:
nit: can we do this after the loop exit? There is no early stop in this loop.

Reply:
> nit: can we do this after the loop exit? There is no early stop in this loop.

Sure, updated, since before exit loop, the last action is `addPartition` so currentGourps if not empty, we don't need to check if currentGroup is empty, just `updateGroups`